app/scripts/controllers/network/network-controller.test.ts

```diff
@@ -7406,658 +6412,1054 @@ describe('NetworkController', () => {
-        await expect(() =>
-          /* @ts-expect-error We are intentionally passing bad input. */
-          controller.upsertNetworkConfiguration({
-            chainId: '0x5',
-            nickname: 'RPC',
-            rpcPrefs: { blockExplorerUrl: 'test-block-explorer.com' },
-            rpcUrl: 'https://mock-rpc-url',
-          }),
-        ).rejects.toThrow(
-          new Error(
-            "Cannot read properties of undefined (reading 'setActive')",
-          ),
+          },
+          async ({ controller }) => {
+            await setFakeProvider(controller, {
+              stubs: [
+                {
+                  request: {
+                    method: 'eth_getBlockByNumber',
+                    params: ['latest', false],
+                  },
+                  response: {
+                    result: {
+                      baseFeePerGas: '0x1',
+                    },
+                  },
+                },
+              ],
+              stubLookupNetworkWhileSetting: true,
+            });
+
+            await operation(controller);
+
+            expect(controller.store.getState().networkDetails).toStrictEqual({
+              EIPS: {
+                1559: true,
+              },
+            });
+          },
         );
       });
     });
 
-    it('should add the given network if all required properties are present but nither rpcPrefs nor nickname properties are passed', async () => {
-      uuidV4Mock.mockImplementationOnce(() => 'networkConfigurationId');
+    it('emits infuraIsUnblocked', async () => {
       await withController(
         {
-          state: {
-            networkConfigurations: {},
-          },
+          state: initialState,
         },
-        async ({ controller }) => {
-          const rpcUrlNetwork = {
-            chainId: '0x1' as const,
-            rpcUrl: 'https://test-rpc-url',
-            ticker: 'test_ticker',
-          };
+        async ({ controller, messenger }) => {
+          await setFakeProvider(controller, {
+            stubLookupNetworkWhileSetting: true,
+          });
 
-          await controller.upsertNetworkConfiguration(rpcUrlNetwork, {
-            referrer: 'https://test-dapp.com',
-            source: MetaMetricsNetworkEventSource.Dapp,
+          const payloads = await waitForPublishedEvents({
+            messenger,
+            eventType: NetworkControllerEventType.InfuraIsUnblocked,
+            operation: async () => {
+              await operation(controller);
+            },
           });
 
-          expect(
-            Object.values(controller.store.getState().networkConfigurations),
-          ).toStrictEqual(
-            expect.arrayContaining([
-              {
-                ...rpcUrlNetwork,
-                nickname: undefined,
-                rpcPrefs: undefined,
-                id: 'networkConfigurationId',
-              },
-            ]),
-          );
+          expect(payloads).toBeTruthy();
```

Comment:
Nit: Do we want to copy over the `toBeFulfilled` matcher? I suppose the diff is easy enough to compare without this, but just curious if you thought about it.

Reply:
Hmm, I do prefer `toBeFulfilled` but maybe it wouldn't be worth the effort? Certainly not in this PR (this pattern is already used heavily outside of the tests in this diff). If you think it'd make it easier for us to make the switch, definitely worth considering though.